src/crh/mod.rs

```diff
@@ -23,3 +25,20 @@ pub trait FixedLengthCRH {
     fn setup<R: Rng>(r: &mut R) -> Result<Self::Parameters, Error>;
     fn evaluate(parameters: &Self::Parameters, input: &[u8]) -> Result<Self::Output, Error>;
 }
+
+pub trait TwoToOneCRH {
+    /// The size for left input and right input
+    const LEFT_INPUT_SIZE_BITS: usize;
+    const RIGHT_INPUT_SIZE_BITS: usize;
+
+    type Output: ToBytes + Clone + Eq + core::fmt::Debug + Hash + Default;
+    type Parameters: Clone + Default;
+
+    fn setup<R: Rng>(r: &mut R) -> Result<Self::Parameters, Error>;
+    fn evaluate(
```

Comment:
```suggestion
	/// Evaluates this CRH on the left and right inputs. 
	/// If `left_input.len() != Self::LEFT_INPUT_SIZE_BITS`, or if
	/// `right_input.len() != Self::RIGHT_INPUT_SIZE_BITS`, then this method returns an error.
    fn evaluate(
```

If the lengths don't match, does this method panic, or does it return an error?

Reply:
I should say panic there. Should I return an error instead?

Reply:
No, that's fine.